ethers-solc/src/lib.rs

```diff
@@ -844,16 +870,22 @@ impl<T: ArtifactOutput + 'static> ProjectCompileOutput<T> {
     /// ```
     pub fn into_artifacts(mut self) -> Box<dyn Iterator<Item = (String, T::Artifact)>> {
         let artifacts = self.artifacts.into_iter().filter_map(|(path, art)| {
-            T::contract_name(&path)
-                .map(|name| (format!("{:?}:{}", path.file_name().unwrap(), name), art))
+            T::contract_name(&path).map(|name| {
+                (format!("{}:{}", path.file_name().unwrap().to_string_lossy(), name), art)
+            })
```

Comment:
Was there a bug here and you do `to_string_lossy`? I guess the issue would be that due to the `{:?}` the key becomes a quoted string, which ends up having a different key from `{}:{}`

Reply:
this added quotes for the file path like `\"file\":name`